source/common/http/conn_manager_impl.cc

```diff
@@ -893,6 +906,13 @@ void ConnectionManagerImpl::ActiveStream::decodeData(ActiveStreamDecoderFilter*
     ENVOY_STREAM_LOG(trace, "decode data called: filter={} status={}", *this,
                      static_cast<const void*>((*entry).get()), static_cast<uint64_t>(status));
 
+    if (!request_metadata_map_vector_.empty()) {
+      for (const auto& metadata_map : request_metadata_map_vector_) {
+        decodeMetadata((*entry).get(), *metadata_map);
```

Comment:
In the request direction, it's more straight-forward to pass the metadata to downstream filters only. But to be consistent with the response direction, shall I pass metadata through all filters? I think we may be able to do something similar on the response direction to enable downstream only filter pass. What do you think?